cmd/monaco/download/download_command.go

```diff
@@ -265,15 +272,30 @@ func setupSharedFlags(cmd *cobra.Command, project, outputFolder *string, forceOv
 // printUploadToSameEnvironmentWarning function may display a warning message on the console,
 // notifying the user that downloaded objects cannot be uploaded to the same environment.
 // It verifies the version of the tenant and, depending on the result, it may or may not display the warning.
-func printUploadToSameEnvironmentWarning(environmentURL, token string) {
-	serverVersion, err := client.GetDynatraceVersion(client.NewTokenAuthClient(token), environmentURL)
+func printUploadToSameEnvironmentWarning(env manifest.EnvironmentDefinition) {
+	var serverVersion version.Version
+	var err error
+	if env.Type == manifest.Classic {
+		serverVersion, err = client.GetDynatraceVersion2ndGen(client.NewTokenAuthClient(env.Auth.Token.Value), env.Url.Value)
+	} else {
+		credentials := client.OauthCredentials{
+			ClientID:     env.Auth.OAuth.ClientId.Value,
+			ClientSecret: env.Auth.OAuth.ClientSecret.Value,
+			TokenURL:     "https://sso-dev.dynatracelabs.com/sso/oauth2/token",
+		}
+		serverVersion, err = client.GetDynatraceVersion2ndGen(client.NewOAuthClient(credentials), env.Url.Value)
```

Comment:
same as the comment in deploy, am I missing something or should this be calling the 3rdGen function?

Reply:
*facepalm* yeah copy paste err